frontend/src/routes/user/login@user.svelte

```diff
@@ -20,49 +18,45 @@
 
 	async function login(): Promise<void> {
 		try {
-			await UserService.login({
-				requestBody: {
-					email: email,
-					password
-				}
-			})
+			const requestBody = {
+				email,
+				password
+			}
+
+			await UserService.login({ requestBody })
+			// Once logged in, we can fetch the workspaces
+			$usersWorkspaceStore = await WorkspaceService.listUserWorkspaces()
+			// And the actual user
+			$userStore = await getUser($workspaceStore!)
+			// Finally, we check whether the user is a superadmin
 			refreshSuperadmin()
+
 			if (rd) {
 				goto(decodeURI(rd))
 			} else {
 				goto('/user/workspaces')
 			}
-		} catch (err) {
-			sendUserToast(`Cannot login: ${err.body}`, true)
+		} catch (error) {
+			sendUserToast(error.message, true)
 		}
 	}
 
 	function handleKeyUp(event: KeyboardEvent) {
 		const key = event.key || event.keyCode
+
 		if (key === 13 || key === 'Enter') {
 			event.preventDefault()
 			login()
 		}
 	}
 
-	async function redirectIfLoggedIn() {
-		try {
-			await UserService.getCurrentEmail()
-			goto('/')
-		} catch {
-			usersWorkspaceStore.set(undefined)
```

Comment:
I do not see this logic anymore but we still need to reset them if a user logout or has an invalid token and is redirected to the login page.

Reply:
Fixed!